### PR TITLE
Keep acronyms/abbreviations in case name when searching for duplicates in Harvard opinion importer

### PR DESCRIPTION
**Pull request by quevon24**
Fix for issue #2204 

With this fix, abbreviations/acronyms won't be removed from case name when searching for case names overlapping.

The problem happens when you have a case name that only contains acronyms/abbreviations, like: “D.L.M. v. T.J.S.”

The current regexes used to find the name overlapping removes the acronyms/abbreviations, in the example above causes to wipe the full string causing duplicates with that type of case names.

With the new regex added, now the acronyms/abbreviations are kept, "D.L.M" turns into "DLM", "U.S." turns into "US", etc.

### CODE REVIEW

File: cl/corpus_importer/management/commands/harvard_opinions.py [hunk 1]
```diff
@@ -967,8 +967,17 @@ def winnow_case_name(case_name: str) -> Set:
...
     case_name = harmonize(case_name)
+
+    # Join abbreviations/acronyms. e.g. "D.L.M. v. T.J.S." -> "DLM v. TJS"
+    case_name = re.sub(
+        r"\b[a-zA-Z][a-zA-Z\.]*[A-Za-z]\b\.?",
+        lambda m: m.group().replace(".", ""),
+        case_name,
+    )
+
+    # Remove all non alphanumeric characters
```
Comment (flooie) [inline cl/corpus_importer/management/commands/harvard_opinions.py:980]: does this remove the chance that we will end up with case names that are blank?

Comment (flooie) [reply cl/corpus_importer/management/commands/harvard_opinions.py:980]: Should we add a check (log an error) if we reduce a title to nothing?

File: cl/corpus_importer/tests.py [hunk 2]
```diff
@@ -681,3 +682,240 @@ def test_short_opinion_matching(self) -> None:
...
+                "dlm",
+                "dem",
+                "child",
+            },
+        )
+        self.assertEqual(
+            winnow_case_name(f"{data_case_1[0]} {data_case_1[1]}"),
+            {
+                "respondents",
+                "minor",
```
Comment (flooie) [inline cl/corpus_importer/tests.py:715]: this function seems particularly long.

Comment (flooie) [reply cl/corpus_importer/tests.py:715]: Can we group all the testing code into a list of tuples (or whatever) at the the start and loop thru it?

Comment (quevon24) [reply cl/corpus_importer/tests.py]: I added some cases in json format, imported them in the test and execute the part of the harvard importer where it checks that the case names don't overlap

File: cl/corpus_importer/management/commands/harvard_opinions.py (hunk 1 above)
Comment (quevon24) [reply cl/corpus_importer/management/commands/harvard_opinions.py:980]: A log could be useful to identify if that happens, I implemented it.

**Comment by mlissner** [comment]
This PR is old and unmerged. What's going on here, @flooie / @quevon24 ?

**Comment by mlissner** [comment]
Feels like this is very relevant to today's PR, no?

File: cl/corpus_importer/tests.py [hunk 3]
```diff
@@ -37,7 +37,7 @@
     OpinionCluster,
     RECAPDocument,
 )
-from cl.settings import INSTALL_ROOT, MEDIA_ROOT
+from cl.settings import MEDIA_ROOT
```
Comment (flooie) [inline cl/corpus_importer/tests.py:40]: nice

**Pull request merged**